### PR TITLE
feat: support searching with hex addr and module path

### PR DESCRIPTION
**Pull request by poomthiti**

### CODE REVIEW

File: src/lib/services/searchService.ts [hunk 1]
```diff
@@ -86,8 +90,35 @@ export const useSearchHandler = (
   const { data: icnsAddressData, isFetching: icnsAddressFetching } =
     useAddressByICNSName(debouncedKeyword);
 
+  const [initiaAddr, moduleName, functionName] = splitModule(debouncedKeyword);
+
+  const enableModuleFetching = useMemo(
+    () =>
+      Boolean(
```
Comment (songwongtp) [inline src/lib/services/searchService.ts:97]: the value is already a boolean no need to cast it again

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Comment by songwongtp** [comment]
@poomthiti @evilpeach 
- should we apply `unpadHexAddress` to the hexaddr result?
- should block height really be searchable via hex or should we prevent it?

**Comment by evilpeach** [comment]
> @poomthiti @evilpeach
> 
> * should we apply `unpadHexAddress` to the hexaddr result?
> * should block height really be searchable via hex or should we prevent it?

1. Yes, applying `unpadHexAddress` to the `hexaddr` would be a good move.
2. It's preferable not to use hex string when searching for block numbers.

File: src/lib/services/searchService.ts (hunk 1 above)
Comment (poomthiti) [reply src/lib/services/searchService.ts:97]: could be `boolean | "" | undefined` if you don't wrap it

**Pull request merged**